### ISSUE TITLE
Formatting Strings where data source contains period/full stop character

### ISSUE THREAD

**Issue by cam-m** [amcharts/amcharts4#2962]
**Question**
I have data with properties that contain a period character. E.g:
```
data = [{
    'Namespaced.Property1': 100,
    'Deeply.Namespaced.Property2': 'Some Text',
    // etc...
},
// etc...
];
```

DataFields seem to accept these properties without issue, however the String Formatting mechanism doesn't handle it as the result is an empty string.

```
Tooltip: {Namespaced.Property1} - {Deeply.Namespaced.Property2}
```
results in:
```
Tooltip:  - 
```

I suspect it has to do with the way the valueX and valueY modifiers function, but is there some way to make this work?

**Comment by martynasma** [comment]
Well, the string formatter treats dots as a hierarchy, meaning it looks for a property "Namespaced" in data item / data context, and if it would be there and it was object, it would then look for "Property1" in it, etc.

For now, there's no workaround, but to replace dot with any other character, e.g. underscore.

We are also going to look if we can enhance formatter to take the string literally. Will let you know if/when it becomes abailable.

**Comment by cam-m** [comment]
Thanks @martynasma that would be great, cheers! (and I didn't know that js object notation worked in string formatter - thanks!)

**Comment by martynasma** [comment]
Fixed in [4.10.5](https://github.com/amcharts/amcharts4/releases/tag/4.10.5).

## [4.10.5] - 2020-10-02

### Changed
- Inline curly-bracket data placeholders now support dot-separated data context names as literal strings, e.g. `"my.custom.property"`.
- A bunch of dependencies were updated to latest versions.

### Fixed
- Toggling `rtl` value on an existing `Label` was (still) not working properly.
- `showSystemTooltip` was being ignored since last version (4.10.4).
- `chart.cursor.snapToSeries` was not causing cursor to snap in wrong place if `groupData` was enabled.
- Bullet tooltip was being hidden even if it had to stay when cursor moved out of the plot area.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by cam-m** [comment]
Thanks @martynasma this worked perfectly - so responsive as always, and much appreciated - C
